### PR TITLE
server: honor container stop timeout from CRI

### PR DESCRIPTION
**Pull request by runcom**
@mrunalp @sameo PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
👍 have we run e2e on this?

**Comment by runcom** [comment]
> have we run e2e on this?

not yet eheh still have to test #508 and #505 :(

File: oci/oci.go [hunk 1]
```diff
@@ -435,29 +435,36 @@ func (r *Runtime) ExecSync(c *Container, command []string, timeout int64) (resp
...
 		return err
 	}
-	i := 0
-	for {
-		if i == 1000 {
-			err := unix.Kill(c.state.Pid, syscall.SIGKILL)
-			if err != nil && err != syscall.ESRCH {
-				return fmt.Errorf("failed to kill process: %v", err)
+	if timeout == -1 {
+		// default 2 seconds delay
```
Comment (mrunalp) [inline oci/oci.go]: Does this match the existing behavior with dockershim?

**Comment by mrunalp** [comment]
Needs rebase.

File: oci/oci.go (hunk 1 above)
Comment (runcom) [inline oci/oci.go]: changed to `10` seconds as it's now in docker https://github.com/moby/moby/blob/master/container/container.go#L53 (not sure why you said "dockershim" though :confused:)

**Comment by runcom** [comment]
@mrunalp rebased

File: oci/oci.go (hunk 1 above)
Comment (mrunalp) [inline oci/oci.go]: Cool :+1:!. That's what everyone is calling the k8s' current CRI over docker :)

**Comment by mrunalp** [comment]
I'll run the e2e on this.

File: oci/oci.go (hunk 1 above)
Comment (runcom) [inline oci/oci.go]: yeah yeah, I thought there was a default stop time _in_ kube but I have clearly misunderstood

**Comment by mrunalp** [comment]
The tests passed fine :+1: LGTM

**Pull request merged**